Frameworks/CoreData/NSEntityDescription.mm

```diff
@@ -14,10 +14,164 @@
 //
 //******************************************************************************
 
+#import <CoreData/CoreData.h>
+#import <Foundation/Foundation.h>
 #import <StubReturn.h>
-#import <CoreData/NSEntityDescription.h>
+
+#import "NSManagedObjectModel-XMLParsing.h"
+
+@interface NSEntityDescription () {
+    NSManagedObjectModel* _managedObjectModel;
+    StrongId<NSArray<NSPropertyDescription*>> _properties;
+    StrongId<NSArray<NSEntityDescription*>> _subentities;
+    StrongId<NSMutableArray<NSPropertyDescription*>> _unresolvedProperties;
+}
+@property (readwrite, assign) NSEntityDescription* superentity;
+@property (readwrite, copy) NSDictionary* subentitiesByName;
+@property (readwrite, copy) NSDictionary* propertiesByName;
+@property (readwrite, copy) NSDictionary* attributesByName;
+@property (readwrite, copy) NSDictionary* relationshipsByName;
+@end
 
 @implementation NSEntityDescription
+- (id)initWithXMLElementName:(NSString*)entityName attributes:(NSDictionary<NSString*, NSString*>*)attributes {
+    if (self = [super init]) {
+        _unresolvedProperties = [NSMutableArray new];
+        _name = [attributes[@"name"] copy];
+        _managedObjectClassName = [attributes[@"representedClassName"] copy];
+        _renamingIdentifier = [attributes[@"elementID"] copy]; // TODO(DH): figure.
```

Comment:
Oh, I think I left it in from before I matched up elementID with renamingIdentifier. I'll cross-check to make sure it's 1) correct and 2) not necessary to have an ambiguous comment